site/docs/spark.md

```diff
@@ -814,3 +815,309 @@ This type conversion table describes how Iceberg types are converted to the Spar
+
+A procedure that rollbacks a table to a certain point in time.
+
+**Note** this procedure invalidates all cached Spark plans that reference the affected table.
+
+#### Usage
+
+| Argument Name | Required? | Type | Description |
+|---------------|-----------|------|-------------|
+| table         | ✔️  | String    | Name of table to rollback |
+| timestamp     | ✔️  | Timestamp | The time to rollback to |
+
+#### Output
+
+| Output Name | Type | Description |
+| ------------|------|-------------|
+| previous_snapshot_id | Long | The current snapshot before rollback |
+| current_snapshot_id  | Long | The new current snapshot id |
+
+#### Example
+
+Rollback `db.sample` to a day ago
+```sql
+    CALL hive_prod.system.rollback_to_timestamp('db.sample', date_sub(current_date(), 1))
+```
+
+### Set Current Snapshot Procedure
+
+Sets the current snapshot for a table to a different snapshot.
+
+**Note** this procedure invalidates all cached Spark plans that reference the affected table.
+
+#### Usage
+
+| Argument Name | Required? | Type | Description |
+|---------------|-----------|------|-------------|
+| table         | ✔️  | String   | Name of table to change |
+| snapshot_id   | ✔️  | Long     | The snapshot to use as current |
+
+#### Output
+
+| Output Name | Type | Description |
+| ------------|------|-------------|
+| previous_snapshot_id | Long | The current snapshot before this procedure |
+| current_snapshot_id  | Long | The new current snapshot id |
+
+#### Example
+
+Set the current snapshot for `db.sample` to 1
+```sql
+    CALL hive_prod.system.set_current_snapshot('db.sample', 1)
+```
+
+### Snapshot Table Procedure
+
+Creates an Iceberg version of a given table without adjusting the underlying table. The new created table can
+be adjusted or written to without adjusting the underlying table. This is basically a test version of
+[migrate](#migrate-table-procedure) which can be used without disrupting users of the original table.
+
+#### Usage
+
+| Argument Name | Required? | Type | Description |
+|---------------|-----------|------|-------------|
+| source_table  | ✔️  | String | Source table for making the Iceberg table |
+| table         | ✔️  | String | Name of the new Iceberg table to create |
+| location      |     | String | Table location for new table (Defaults to catalog's default location) |
+| properties    | ️   | Map<String, String> | Properties to add to the newly created table |
+
+#### Output
+
+| Output Name | Type | Description |
+| ------------|------|-------------|
+| imported_files_count | Long | Number of files referenced by the new table |
+
+#### Examples
+
+Make an isolated Iceberg table which references table `db.sample` named `db.snap` at the
+catalog's default location for `db.snap`.
+```sql
+    CALL catalog_name hive_prod.system.snapshot('db.sample', 'db.snap')
+```
+
+Migrate an isolated Iceberg table which references table `db.sample` named `db.snap` at
+a manually specified location `/tmp/temptable/`.
```

Comment:
This location is not a valid URI. In this case, will it choose the URI scheme of the source table?

Reply:
No, it basically sets the Iceberg Table location. 
```
Catalog.TableBuilder.withLocation("/tmp/temptable/")
```
Which will set the "location" parameter in TableMetadata, I think normally this is going to be resolved as a HDFS path so it would be the default scheme ('default.fs'), but I think it may be possible for a custom FileIO to resolve it differently.

Reply:
I see. In this case I think it is preferable to use a URI in the examples, because not all FileIO can resolve a non-URI path correctly.